Bottom: 31ec666869dc5b2d9a5f6df475af597343469a74
Top:    bf57fb5f274ac72c9618cedf1dbf04f69a9b4c80
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-12 01:25:37 +0100

More hack to avoid GL calls when we're out of context


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 59c5fd8..a152fae 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -506,6 +506,12 @@ ghid_set_color (hidGC gc, const char *name)
 
   gc->colorname = (char *) name;
 
+  if (!check_gl_drawing_ok_hack)
+    {
+      current_color = NULL;
+      return;
+    }
+
   if (gport->colormap == NULL)
     gport->colormap = gtk_widget_get_colormap (gport->top_window);
   if (strcmp (name, "erase") == 0)
